lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1140,6 +1139,12 @@ WasmBytecodeGenerator::EmitMemAccess(WasmOp wasmOp, const WasmTypes::WasmType* s
     {
         throw WasmCompilationException(_u("Index expression must be of type I32"));
     }
+
+    if (offset >= m_module->GetMemory()->maxSize)
+    {
+        throw WasmCompilationException(_u("Index is out of bounds"));
```

Comment:
yeah, we probably need to emit different code here. if index+offset overflow uint32, then we need to trap at runtime. this check doesn't prevent that from happening. For example, even with small offset you might have some large index that causes index+offset to overflow

---
In reply to: [85627647](https://github.com/Microsoft/ChakraCore/pull/1823#discussion_r85627647) [](ancestors = 85627647)